pyanaconda/kickstart.py

```diff
@@ -1070,6 +1070,19 @@ def execute(self, *args):
             logger.updateRemote(remote_server)
 
 class Network(commands.network.RHEL7_Network):
+    def parse(self, args):
+        nd = commands.network.RHEL7_Network.parse(self, args)
+        setting_only_hostname = nd.hostname and len(args) <= 2
```

Comment:
Args are either
["--hostname=blah']
or
["--hostname", "blah"] for
network --hostname blah